Update tensornet backends docs (#871)
NVIDIA/cuda-quantum: docs/sphinx/using/simulators.rst

This backend exposes a set of environment variables to configure specific aspects of the simulation:

* **`CUDAQ_MPS_MAX_BOND=X`**: The maximum number of singular values to keep (fixed extent truncation). Default: 64.
* **`CUDAQ_MPS_ABS_CUTOFF=X`**: The cutoff for the largest singular value during truncation. Eigenvalues that are smaller will be trimmed out. Default: 1e-5.
* **`CUDAQ_MPS_RELATIVE_CUTOFF=X`**: The cutoff for the maximal singular value relative to the largest eigenvalue. Eigenvalues that are smaller than this fraction of the largest singular value will be trimmed out. Default: 1e-5

The :code:`tensornet-mps` only supports single-GPU simulation. Its approximate nature allows the :code:`tensornet-mps` backend to handle a large number of qubits for certain classes of quantum circuits on a relatively small memory footprint.

.. warning:: 

>     The :code:`tensornet-mps` cannot handle quantum gates acting on more than two qubit operands. It will throw an error when this constraint is not satisfied.

Reviewer: I thought I also saw a similar commend and some disabled tests for the tensornet backend. If so, there should be a similar note in the section above for the `tensornet` target.

Author: We disabled those tests involving 3+ qubit gates only for the `tensornet-mps` backend.
We also disabled tests involving checking the random seed and tests that have long runtime for both backends. 
I've added a [ticket](https://github.com/NVIDIA/cuda-quantum/issues/885) to address this.

Usage
++++++

To specify the use of the :code:`tensornet` or :code:`tensornet-mps` target, pass the following command line 
options to :code:`nvq++`

.. code:: bash 

    nvq++ --target tensornet src.cpp ...

or 

.. code:: bash 

    nvq++ --target tensornet-mps src.cpp ...

> Similarly, when using Python, the :code:`--target` command line option can also be used

Reviewer: Instead of separately listing code for C++ and Python, please use tabs for showing the code in the selected language instead, the way it is done e.g. here: https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/api/default_ops.rst

Author: Fixed.

Tensor Network Simulators
==================================

CUDA Quantum provides a couple of tensor-network simulator targets accelerated with 
the :code:`cuTensorNet` library. 
These backends are available for use from both C++ and Python.

> `cuTensorNet` multi-node multi-GPU

Reviewer 2: Capitalization of multi-node multi-GPU is inconsistent with line 92. I would recommend making them consistent or capitalizing the one in line 87 since it is a subject heading and uncapitalizing the one in line 92. My GitHub suggestions are for the latter, but I'm ok with either way.
```suggestion
`cuTensorNet` Multi-Node Multi-GPU
```

+++++++++++++++++++++++++++++++++++

The :code:`tensornet` backend represents quantum states and circuits as tensor networks in an exact form (no approximation). 
Measurement samples and expectation values are computed via tensor network contractions. 

> This backend supports Multi-Node, Multi-GPU distribution of tensor operations required to evaluate and simulate the circuit.

Reviewer 2: ```suggestion
This backend supports multi-node, multi-GPU distribution of tensor operations required to evaluate and simulate the circuit.
```

.. note:: 

>     To enable automatic distributed parallelization across multiple/many GPUs for the :code:`tensornet` backend, users need to activate `cuTensorNet`'s distributed interface 

Reviewer: ```suggestion
    To enable automatic distributed parallelization across multiple/many GPUs for the :code:`tensornet` backend, `cuTensorNet`'s distributed interface needs to be activated
```

* **`OMP_PLACES=cores`**: Set this environment variable to improve CPU parallelization.
* **`OMP_NUM_THREADS=X`**: To enable CPU parallelization, set X to `NUMBER_OF_CORES_PER_NODE/NUMBER_OF_GPUS_PER_NODE`.

.. note:: 

    The **CUDA_VISIBLE_DEVICES** environment variable should **always** be set before using the :code:`tensornet` 
    backend if you have multiple GPUs available. With OpenMPI, you can run a multi-GPU quantum circuit simulation like this:

    .. code:: bash 
    
        mpiexec -n 8 sh -c 'CUDA_VISIBLE_DEVICES=${OMPI_COMM_WORLD_LOCAL_RANK} binary.x > tensornet.${OMPI_COMM_WORLD_RANK}.log'

    This command will assign a unique GPU to each MPI process within the node with 8 GPUs and produce a separate output for each MPI process.

> `cuTensorNet` matrix product state 

Reviewer 2: Suggest capitalizing the subject heading
```suggestion
`cuTensorNet` Matrix Product State 
```
